BUG: Add warning when removing empty columns

ElDeveloper opened the pull request:
A UserWarning is now raised whenever load_template_to_dataframe drops a column
if it is empty. This message is shown on the interface as well.

Fixes #740

antgonza commented on qiita_db/test/test_metadata_template.py:
```diff
@@ -1481,6 +1482,21 @@ def test_load_template_to_dataframe_empty_sample_names(self):
 
+    def test_load_template_to_dataframe_empty_column(self):
+
+            obs = npt.assert_warns(UserWarning, load_template_to_dataframe,
```
Block is over indented, interesting that flake8 didn't complain.

antgonza commented on qiita_pet/handlers/study_handlers/description_handlers.py:
```diff
@@ -280,9 +295,20 @@ def add_prep_template(self, study, user, callback):
+                # will be ignored if an exception is raised
+                if warns:
+                    msg = '; '.join([str(w.message) for w in warns])
+                    msg_level = 'danger'
```
If this is a warning, should it be warning level vs. error?

antgonza replied:
While testing I realized that danger is actually the warning color so ignore.

antgonza commented:
One comment that is not blocking so :+1:

ElDeveloper commented:
I've addressed your comments, thanks! Note that the level was wrong,
should have been warning, not danger. Not sure why did it show you the
correct level. Maybe your browser cached the CSS value?

On (Feb-05-15| 4:43), Antonio Gonzalez wrote:
> One comment that is not blocking so :+1: 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/823#issuecomment-73040182

ElDeveloper commented:
If someone else can review this PR, it should be ready for merge.

josenavas commented:
@ElDeveloper which is the expected color of the message? From the code, it looks like it should be yellow (`warning`). However, when it shows up as `info` level (light blue):
![screen shot 2015-02-05 at 5 44 27 pm](https://cloud.githubusercontent.com/assets/2501478/6073121/db3412d2-ad5e-11e4-8707-b2f36656a895.png)

josenavas commented:
Looks like the problem is [here] (https://github.com/biocore/qiita/blob/master/qiita_pet/templates/sitebase.html#L101) in sitebase.html

ElDeveloper commented:
Thanks for catching that, I tested it but did not pay much attention to
the color and ended up blaming my browser. I'll fix this.

On (Feb-05-15|17:54), josenavas wrote:
> Looks like the problem is [here] (https://github.com/biocore/qiita/blob/master/qiita_pet/templates/sitebase.html#L101) in sitebase.html
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/823#issuecomment-73167928

josenavas commented:
Thanks @ElDeveloper It works perfect now:
![screen shot 2015-02-05 at 6 20 57 pm](https://cloud.githubusercontent.com/assets/2501478/6073483/d518c0d2-ad63-11e4-9c92-49ab0c138ef8.png)

ElDeveloper commented:
Thanks!

On (Feb-05-15|18:22), josenavas wrote:
> Merged #823.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/823#event-231218632
